### ISSUE TITLE
Doesn't work with cmd on windows 10

### ISSUE THREAD

**Issue by wald-tq** [firsttris/vscode-jest-runner#53]
Right clicking on a test name and selecting "Run Jest" doesn't work with windows 10 and cmd as default shell.

following error message appears:

```
E:\dev\EM Cloud\cloud-portal\backend>cd "e:\dev\EM Cloud\cloud-portal\backend"; node "e:/dev/EM Cloud/cloud-portal/backend/node_modules/jest/bin/jest.js" "e:/dev/EM Cloud/cloud-portal/backend/tests/login.test.ts" -t "Register new user successful"
Die Syntax für den Dateinamen, Verzeichnisnamen oder die Datenträgerbezeichnung ist falsch.
```

It works using Powershell or Git bash as default shell.

**Comment by entranced** [comment]
'Debug jest' and 'Run jest' don't seem to be working in cmd or Powershell. 
'Run Jest File' works with Powershell.

**Comment by firsttris** [comment]
hey, the slashes of everything what comes after node .... are incorrect for windows.
this worked once.

Will fix this issue soon.

in the mean time just use powershell :)

**Comment by yaagma** [comment]
Hello. I think it would be enough to replace the ';' with '&'.

For example, the command generated by "Run jest" is:
```
cd "c:\..." ; node "c:/.../jest.js" "c:/...main.test.ts" -t "^check result value$"
```
It works if I change ';' with '&':
```
cd "c:\..." & node ...
```

I hope it helps.
Raúl

**Comment by firsttris** [comment]
does && work with cmd as well? this would also work for bash.

**Comment by yaagma** [comment]
Yes, it works, but it is not exactly the same as `&`. With `&&`, the second command only is executed if the first doesn't fail:
[From Microsoft docs](https://docs.microsoft.com/en-us/previous-versions/windows/it-pro/windows-xp/bb490954(v=technet.10)):

Character|Definition
---|---
&  | Use to separate multiple commands on one command line. Cmd.exe runs the first command, and then the second command.
&& | Use to run the command following && only if the command preceding the symbol is successful. Cmd.exe runs the first command, and then runs the second command only if the first command completed successfully.

I think that, for "Run Jest", the difference doesn't matter and `&&` could be used.

Thanks for this great extension.
Raúl

Ps: By the way. ¿Are codelens for "Debug Jest"/"Run Jest" in your plans?

**Comment by wald-tq** [comment]
> Ps: By the way. ¿Are codelens for "Debug Jest"/"Run Jest" in your plans?

That would be cool!

**Comment by firsttris** [comment]
codelens option would be indeed nice :+1:  

i have created another issue for this feature: https://github.com/firsttris/vscode-jest-runner/issues/56
if you have some more information on this topic (like API details), feel free to post it there.

i want to fix the remaining issues first.

but if you are able to implement something like this, im happy about any PR.

**Comment by heavenkiller2018** [comment]
in the mean time just use powershell :)  yes, i got it !

**Comment by firsttris** [comment]
please verify and report back if your issue is solved with 0.4.12

**Comment by firsttris** [comment]
changed ; to &&

**Comment by wald-tq** [comment]
Hi Tristan,

it works with cmd but not with powershell:
```
The token '&&' is not a valid statement separator in this version.
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : InvalidEndOfLine
```

**Comment by firsttris** [comment]
fixed it with 0.4.15. it seems && is supported by cmd, bash, fish but not powershell. its harder then you think to support all terminals also considering multi workspace setups and mono repos.

feel free to re-open or create a new issue if you find one.
